code/datums/traits/negative_quirk.dm

```diff
@@ -426,8 +426,11 @@
 	var/slot_string = "limb"
 
 /datum/quirk/prosthetic_limb/on_spawn()
-	var/limb_slot = pick(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
+	var/limb_slot = pick(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG) // default to random
 	var/mob/living/carbon/human/H = quirk_target
+	var/client/C = GLOB.directory[ckey(quirk_holder.key)] // this was the only way I could find to get the client during on_spawn()
```

Comment:
does `mind.current.client` not work?  or `quirk_target.client`?

Also put `ckey(quirk_holder.key)` into a var before passing it to directory

Reply:
both of those are null at the time of on_spawn(), I tried using those first

Reply:
put it into a var :3